root/usr/lib/node/nethcti-server/plugins/com_astproxy_rest/plugins_rest/astproxy.js

```diff
@@ -2728,6 +2731,58 @@ var compConfigManager;
         }
       },
 
+            /**
+       * Cancel a call registered with a supported
+       * physical phone with the following REST API:
+       *
+       *     POST cancel
+       *
+       * @method call
+       * @param {object} req The client request
+       * @param {object} res The client response
+       * @param {function} next Function to run the next handler in the chain
+       */
+            cancel: function (req, res, next) {
+              try {
+                var username = req.headers.authorization_user;
+                var result = compUser.getUserInfoJSON(username);
+                const nethlinkExtensions = result.endpoints[compUser.ENDPOINT_TYPES.extension].filter((endpoint) => endpoint.type === 'nethlink');
+                const nethlinkExtension = nethlinkExtensions.length > 0 ? nethlinkExtensions[0].id : null;
+                var nethlinkStatus = compAstProxy.getExtenStatus(nethlinkExtension);
+
+                // check parameters
+                if (typeof req.params !== 'object' || typeof req.params.endpointId !== 'string') {
+                  compUtil.net.sendHttp400(IDLOG, res);
+                  return;
+                }
+
+                // check if the extension of the request is owned by the user: the user
+                // can only cancel a conversation that belong to him
+                if (compAuthorization.verifyUserEndpointExten(username, req.params.endpointId) === false) {
+
+                  logger.log.warn(IDLOG, 'cancel call from "' + username + '" has been failed: the extension "' +
+                    req.params.endpointId + '" is not owned by him');
+                  compUtil.net.sendHttp403(IDLOG, res);
+                  return;
+                }
+
+                var extenAgent = compAstProxy.getExtensionAgent(req.params.endpointId);
+                var isSupported = compConfigManager.phoneSupportHttpApi(extenAgent);
+
+                if (!isSupported) {
+                  var str = 'cancel conversation with unsupported phone (exten: ' + req.params.endpointId + '/' + extenAgent + ')';
+                  logger.log.warn(IDLOG, str);
+                  compUtil.net.sendHttp500(IDLOG, res, str);
+                } else if (isSupported && compAstProxy.isAutoC2CEnabled && nethlinkStatus === 'online') {
+                  sendCancelConversation(username, req, res);
+                  compUtil.net.sendHttp200(IDLOG, res);
+                }
```

Comment:
what happen if device is supported but compAstProxy.isAutoC2CEnabled is false or nethlinkStatus isn't online?

Reply:
There is a fallback to manual click2call